### PR TITLE
@curve/address provider ownership transfer

### PR DESCRIPTION
**Pull request by yasmine9**
Create Address-provider-ownership-transfer Agent using the  agent template.

### CODE REVIEW

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Supported Chains
+
+- Ethereum
+
+## Alerts
+
+Describe each of the type of alerts fired by this agent
+
+- CURVE-4
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/README.md:15]: Looks like you are using `CURVE-4` here and `CURVE-3` when creating the handler in `agent.ts`

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  return (metadata: { [key: string]: any } | undefined) => {
+    return Finding.fromObject({
+      name: "Curve Admin Event Detected",
+      description: "An ownership transfer has been initiated",
+      alertId: alertID,
+      severity: FindingSeverity.Medium,
+      type: FindingType.Info,
+      metadata: {
+        newAdmin: metadata!.topics[2],
+      },
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:27]: Could we add 
`protocol: Curve Finance`?

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  },
+});
+
+describe("Transfer Ownership Agent for Registry Contract", () => {
+  let handleTransaction: HandleTransaction = provideCommitNewAdminEvent(ALERT_ID, ADDRESS);
+  let topics1: string[];
+  let topics2: string[];
+
+  beforeAll(() => {
+    topics1= [DURATION, NEW_ADMINS[0]];
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.spec.ts:39]: Could you run a formatter for avoiding this?

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  FindingType,
+  HandleTransaction,
+  ethers,
+} from "forta-agent";
+import {
+  FindingGenerator,
+  provideEventCheckerHandler,
+} from "forta-agent-tools";
+
+export const ADDRESS = "0x0000000022d53366457f9d5e68ec105046fc4383"; // Address Provider contract address
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:13]: Maybe we can do the same check on other contracts too.

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  address: string
+): HandleTransaction => provideEventCheckerHandler(
+      createFindingGenerator(alertID),
+      iface.getEvent('CommitNewAdmin').format('sighash'),
+      address
+    );
+
+export default {
+  handleTransaction: provideCommitNewAdminEvent(
+    "CURVE-3",
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:43]: Can we change this to `CURVE-1` for following the order from the table at `https://www.notion.so/nethermind/CurveDAO-7807d9f852ac42f2973c5aa4747a80af`

Comment (e1Ru1o) [reply Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:43]: We are following the order of the **Agent** section instead of table

Comment (mauricio1802) [reply Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:43]: Ohh I see, but looks like not all of the agents are there right? The table has much more

Comment (e1Ru1o) [reply Curve-finance-suite/Address-Provider-Ownership-Transfer/src/agent.ts:43]: Yes, once you pick an agent, you need to write your agent in the list, pick the next available Id and also wrote the proposed solution.

**Review by e1Ru1o — APPROVED** [review]
LGTM only needs to resolve the current comments

**Review by mauricio1802 — APPROVED** [review]

**Pull request merged**